addons/mass_mailing/controllers/main.py

```diff
@@ -92,8 +93,11 @@ def unsubscribe(self, mailing_id, opt_in_ids, opt_out_ids, email, res_id, token)
         return 'error'
 
     @http.route('/mail/track/<int:mail_id>/blank.gif', type='http', auth='public')
-    def track_mail_open(self, mail_id, **post):
+    def track_mail_open(self, mail_id, token, **post):
         """ Email tracking. """
+        if not consteq(token, sign_data(request.env, ('blank.gif', mail_id))):
```

Comment:
Don't authenticate a tuple, should be a string. Any why do you append a static string to the message anyway?

Reply:
@mart-e To have some "salt" linked to the usage, otherwise the token used for the blank gif of the mail<id=5> will be the same as the token used in an other module, for a record<id=5>, if you see what I mean

Reply:
How about a classic `model,res_id` construction style? Or a more explicative string like "email-tracking-id-%s-opened". Again, don't need to be a tuple.